### PR TITLE
Update to tx tracking and tests

### PR DESCRIPTION
**Pull request by codyborn**
- Rely solely on `checkTransaction` process for transaction state tracking
- Update tests with more checks

### CODE REVIEW

File: apps/relayer/src/transaction/transaction.service.ts [hunk 1]
```diff
@@ -113,24 +124,6 @@ export class TransactionService implements OnModuleInit, OnModuleDestroy {
   private watchTransaction(txHash: string, result?: TransactionResult) {
     this.watchedTransactions.add(txHash)
     this.transactionSeenAt.set(txHash, Date.now())
-
-    if (result !== undefined) {
-      this.waitForReceipt(result).catch(e => {
```
Comment (bowd) [inline apps/relayer/src/transaction/transaction.service.ts:118]: Yeah, I dunno why I insisted on using the receipt here and not just relied on the overall polling 😅
Also I guess you can remove the optional `result` argument of the function.

Comment (codyborn) [reply apps/relayer/src/transaction/transaction.service.ts:118]: Yeah, I think it's a bit cleaner to just rely on a single mechanism to update the state. Good point on the `result` field.. removed!

File: apps/relayer/src/transaction/transaction.service.spec.ts [hunk 2]
```diff
@@ -188,11 +194,29 @@ describe('TransactionService', () => {
...
+        })
+
+        // Simulate being included in a block and ensure it's unwatched
+        const completedTx = txFixture()
+        completedTx.hash = tx.hash
+        completedTx.blockHash = "notNull"
+        const completedTxPromise = Promise.resolve(completedTx)
+        getTransaction.mockReturnValue(completedTxPromise)
+        jest.advanceTimersToNextTimer(2)
+        await completedTxPromise
```
Comment (codyborn) [inline apps/relayer/src/transaction/transaction.service.spec.ts:215]: I noticed that the async calls within `checkTransactions` wasn't being executed until I called await explicitly here

Comment (codyborn) [reply apps/relayer/src/transaction/transaction.service.spec.ts:215]: This describes it pretty well:
https://github.com/facebook/jest/issues/9181#issuecomment-567384171

**Review by bowd — APPROVED** [review]
LGTM

**Pull request merged**